core/commonTypes.ml

```diff
@@ -0,0 +1,111 @@
+module Linearity = struct
+  type t = Any | Unl
+    [@@deriving eq,show]
+
+  let isAny lin = lin == Any
+  let isUnl lin = lin == Unl
+
+  let string_of = function
+    | Any -> "Any"
+    | Unl -> "Unl"
+end
+
+(* Convenient aliases for constructing values *)
+let linUnl = Linearity.Unl
+let linAny = Linearity.Any
+
+module DeclaredLinearity = struct
+  type t = Lin | Unl
+    [@@deriving show]
+
+  let isLin lin = lin == Lin
```

Comment:
`is_lin` (maybe even `is_linear`)

Reply:
You are using reference equality `==`, I think you should use structural equality `=`. Even better use pattern matching, that shouldn't use the polymorphic equality operator, e.g.
```ocaml
let is_lin =  function Lin -> true | _ -> false
```